code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -370,21 +376,46 @@
 		src.antigrab_counter = 0
 	if(keys & KEY_RUN)
 		if(!src.floorrunning && isfeathertile(src.loc))
-			if(istype(src.loc, /turf/simulated/floor/feather))
+			if (length(src.grabbed_by))
+				for(var/obj/item/grab/g in src.grabbed_by)
+					if (!(g.state == GRAB_PASSIVE || g.state == GRAB_PIN)) // in the rare case you do pin a flockdrone
+						src.can_floorrun = FALSE
+						return ..()
+			else if (!src.can_floorrun)
+				src.can_floorrun = TRUE
```

Comment:
seems redundant

Reply:
I just removed the if check if that's what you were meaning